Sources/SwiftIfConfig/VersionTuple+Parsing.swift

```diff
@@ -0,0 +1,105 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+import SwiftSyntax
+
+extension VersionTuple {
+  /// Parse a compiler build version of the form "5007.*.1.2.3*", which is
+  /// used by an older if configuration form `_compiler_version("...")`.
+  /// - Parameters:
+  ///   - versionString: The version string for the compiler build version that
+  ///   we are parsing.
+  ///   - versionSyntax: The syntax node that contains the version string, used
+  ///   only for diagnostic purposes.
+  init(
+    parsingCompilerBuildVersion versionString: String,
+    _ versionSyntax: ExprSyntax
+  ) throws {
+    components = []
+
+    // Version value are separated by periods.
+    let componentStrings = versionString.split(separator: ".")
+
+    /// Record a component after checking its value.
+    func recordComponent(_ value: Int) throws {
+      let limit = components.isEmpty ? 9223371 : 999
```

Comment:
Where is 9223371 coming from? The value seems pretty arbitrary to me…

Reply:
Stolen straight out of the Swift compiler code base without thinking ;)